segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/LogMetadata.java

```diff
@@ -140,6 +141,63 @@ LogMetadata truncate(LedgerAddress upToAddress) {
                 .withUpdateVersion(this.updateVersion.get());
     }
 
+    /**
+     * Removes LedgerMetadatas for those Ledgers that are known to be empty.
+     *
+     * @param skipCountFromEnd The number of Ledgers to spare, counting from the end of the LedgerMetadata list.
+     * @return A new instance of LogMetadata with the updated ledger list.
+     */
+    LogMetadata removeEmptyLedgers(int skipCountFromEnd) {
```

Comment:
Just so that I understand, what's the reason for having this argument rather than directly use `Ledgers.MIN_FENCE_LEDGER_COUNT` to compute the cut-off?

Reply:
I didn't want LogMetadata to have a dependency on Ledgers.

Reply:
ok.